services/geocoding.py

```diff
@@ -27,15 +24,21 @@ def enhance(self, entries: List[PointOfInterest]) -> List[PointOfInterest]:
         For entries with missing lat/lng but present address
         we can find lat/lng using a geocoding API.
         """
+        #Create client instance of gmaps
+        gmaps = googlemaps.Client(key=self.settings.developer_key)
+
         addresses_to_geocode: List[str] = [
             poi.address
             for poi in entries
             if not poi.lat or not poi.lng
         ]
 
         coordinates: Mapping[str, Point] = {}
-        # TODO: use a geocoding API to fill the coordinates for each address
 
+        for address in addresses_to_geocode:
```

Comment:
Do they support bulk queries? I'd rather make just one request than many in a loop

Reply:
No, unfortunately not but this reminded me to check if there are more than 50 addresses because you can only submit 50 a second and then I have to add a await. You can get that service from other providers like ArcGIS, Here, and MapQuest.

Reply:
Maybe we should use one of those? Running a bunch of requests sequentially is not ideal if there was an option to do it in bulk...

Reply:
These ones support batch and have a Python SDK:
* https://developers.arcgis.com/python/
* https://github.com/bennylope/pygeocodio

Reply:
These ones support batch and have a Python SDK:
* https://developers.arcgis.com/python/
* https://github.com/bennylope/pygeocodio